### PR TITLE
Registration Smoke

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
Automate Registration Smoke according to template
https://adc.luxoft.com/jira/browse/SDLOPEN-916 (see attached files)

### CODE REVIEW

**Comment by LitvinenkoIra** [comment]
@LuxoftAKutsan @VVeremjova @aderiabin @dboltovskyi please review

File: test_scripts/Smoke/Registration/ATF_Register_5_connection.lua [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        commonFunctions:userPrint(35, "HMI is ready")
+      end)
+    end)
+  end)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Check that it is able to register up to 5 Apps on different connections")
+
+function Test:FirstConnection()
```
Comment (VVeremjova) [inline test_scripts/Smoke/Registration/ATF_Register_5_connection.lua]: @LitvinenkoIra How about create function which will have arguments that are different for each connection?

Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Register_5_connection.lua]: @VVeremjova Done
8177595051e65da23928d90c8aa45b24762737a9

File: test_scripts/Smoke/Registration/ATF_Register_5_connection.lua [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  self.session_name = mobile_session.MobileSession(self, self.connection_name, app)
+  event_dispatcher:AddConnection(self.connection_name)
+  self.session_name:ExpectEvent(events.connectedEvent, "Connection started")
+  self.connection_name:Connect()
+  self.session_name:StartService(7):Do(function()
+    local correlationId = self.session_name:SendRPC("RegisterAppInterface", app)
+    EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered", {application = { appName = app.appName}})
+    self.session_name:ExpectResponse(correlationId , { success = true, resultCode = "SUCCESS"})
+    self.session_name:ExpectNotification("OnHMIStatus",{hmiLevel = "NONE", 
+          audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN"}):Do(function(_, data)
```
Comment (dboltovskyi) [inline test_scripts/Smoke/Registration/ATF_Register_5_connection.lua]: @LitvinenkoIra, Parameter "data" is unused and can be removed

File: test_scripts/Smoke/Registration/ATF_Register_5_connection.lua [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+      end)
+    end)
+  end)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Check that it is able to register up to 5 Apps on different connections")
+
+function Test:FirstConnection()
+  createConnectionAndRegisterApp(self, device1, "mobile1.out", default_app_params1, mobileConnection1, mobileSession1)
```
Comment (dboltovskyi) [inline test_scripts/Smoke/Registration/ATF_Register_5_connection.lua]: @LitvinenkoIra, Here and below parameters "mobileConnection1" and "mobileSession1" are undefined

File: test_scripts/Smoke/Registration/ATF_Register_5_connection.lua [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+-- [[Local variables]]
+local default_app_params1 = config.application1.registerAppInterfaceParams
+local default_app_params2 = config.application2.registerAppInterfaceParams
+local default_app_params3 = config.application3.registerAppInterfaceParams
+local default_app_params4 = config.application4.registerAppInterfaceParams
+local default_app_params5 = config.application5.registerAppInterfaceParams
+local devicePort = 12345
+
+--[[ Local Functions ]]
+local function createConnectionAndRegisterApp(self, device, filename, app, connection_name, session_name)
```
Comment (dboltovskyi) [inline test_scripts/Smoke/Registration/ATF_Register_5_connection.lua]: @LitvinenkoIra, Parameters "connection_name" and  "session_name" are unused

File: test_scripts/Smoke/Registration/ATF_Register_5_session.lua [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+require('user_modules/AppTypes')
+
+-- [[Local variables]]
+local default_app_params2 = config.application2.registerAppInterfaceParams
+local default_app_params3 = config.application3.registerAppInterfaceParams
+local default_app_params4 = config.application4.registerAppInterfaceParams
+local default_app_params5 = config.application5.registerAppInterfaceParams
+
+--[[ Local Functions ]]
+local function startSessionAndRegisterApp(self, session_name, app)
```
Comment (dboltovskyi) [inline test_scripts/Smoke/Registration/ATF_Register_5_session.lua]: @LitvinenkoIra, Parameter "session_name" is unused

File: test_scripts/Smoke/Registration/ATF_Register_5_session.lua [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+      end)
+    end)
+  end)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Check that it is able to register 5 sessions within 1 phisycal connection")
+
+function Test:Start_Session2_And_Register_App_2()
+  startSessionAndRegisterApp(self, mobileSession2, default_app_params2)
```
Comment (dboltovskyi) [inline test_scripts/Smoke/Registration/ATF_Register_5_session.lua]: @LitvinenkoIra, Here and below "mobileSession2" parameter is undefined.

File: test_scripts/Smoke/Registration/ATF_Register_App.lua [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+--  Mobile device is connected to system.
+--
+--  2. Performed steps
+--  appID->RegisterAppInterface(params)
+--
+--  Expected behavior:
+--  1. SDL successfully registers application and notifies HMI and mobile
+--     SDL->HMI: OnAppRegistered(params)
+--     SDL->appID: SUCCESS, success:"true":RegisterAppInterface()
+--  3. SDL assignes HMILevel after application registering:
```
Comment (dboltovskyi) [inline test_scripts/Smoke/Registration/ATF_Register_App.lua]: @LitvinenkoIra, Please correct expectations numbering (2nd is missing)

File: test_scripts/Smoke/Registration/ATF_Register_App.lua [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+-- [[Local variables]]
+local default_app_params = config.application1.registerAppInterfaceParams
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+commonSteps:DeletePolicyTable()
+commonSteps:DeleteLogsFiles()
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Check that it is able to register App")
```
Comment (dboltovskyi) [inline test_scripts/Smoke/Registration/ATF_Register_App.lua]: @LitvinenkoIra, Please correct description in following way (according to Atomic_template): 
```
commonFunctions:newTestCasesGroup("Test")
```

File: test_scripts/Smoke/Registration/ATF_Register_App.lua [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+end
+
+function Test:Register_App()
+  self.mobileSession = mobile_session.MobileSession(self, self.mobileConnection)
+  local on_rpc_service_started = self.mobileSession:StartRPC()
+  on_rpc_service_started:Do(function()
+    local correlation_id = self.mobileSession:SendRPC("RegisterAppInterface", default_app_params)
+    EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered", { application = { appName = default_app_params.appName}})
+
+    EXPECT_RESPONSE(correlation_id, {success = true, resultCode = "SUCCESS"}):Do(function(_,data)
```
Comment (dboltovskyi) [inline test_scripts/Smoke/Registration/ATF_Register_App.lua]: @LitvinenkoIra, Parameter "data" is unused

File: test_scripts/Smoke/Registration/ATF_Reregister_App.lua [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+--  2. SDL successfully registers application and notifies HMI and mobile
+--     SDL->HMI: OnAppRegistered(params)
+--     SDL->appID: SUCCESS, success:"true":RegisterAppInterface()
+--  3. SDL assignes HMILevel after application registering:
+--     SDL->appID: OnHMIStatus(HMlLevel, audioStreamingState, systemContext)
+
+-- [[ Required Shared Libraries ]]
+local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local mobile_session = require('mobile_session')
```
Comment (dboltovskyi) [inline test_scripts/Smoke/Registration/ATF_Reregister_App.lua]: @LitvinenkoIra, "mobile_session" is unused

File: test_scripts/Smoke/Registration/ATF_Reregister_App.lua [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+            commonFunctions:userPrint(35, "App is registered")
+          end)
+        end)
+      end)
+    end)
+  end)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Check that it is able to reregister App within current connection")
```
Comment (dboltovskyi) [inline test_scripts/Smoke/Registration/ATF_Reregister_App.lua]: @LitvinenkoIra, Please correct description in following way (according to Atomic_template): 
```
commonFunctions:newTestCasesGroup("Test")
```

File: test_scripts/Smoke/Registration/ATF_Reregister_App.lua [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+function Test:Unregister_App()
+  local cid = self.mobileSession:SendRPC("UnregisterAppInterface", default_app_params)
+  EXPECT_RESPONSE(cid, { success = true, resultCode = "SUCCESS"})
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", {unexpectedDisconnect = false, 
+                   appID = self.applications[default_app_params]})
```
Comment (dboltovskyi) [inline test_scripts/Smoke/Registration/ATF_Reregister_App.lua]: @LitvinenkoIra, By default, when app is registered appId is stored in self.applications[app.appName]. Therefore self.applications[default_app_params] will return nil

File: test_scripts/Smoke/Registration/ATF_Reregister_App.lua [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+end
+
+function Test:Reregister_Application()
+  local correlation_id = self.mobileSession:SendRPC("RegisterAppInterface", default_app_params)
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered", { application = { appName = default_app_params.appName}})
+
+  EXPECT_RESPONSE(correlation_id, {success = true, resultCode = "SUCCESS"}):Do(function(_,data)
```
Comment (dboltovskyi) [inline test_scripts/Smoke/Registration/ATF_Reregister_App.lua]: @LitvinenkoIra, Parameter "data" is unused

File: test_scripts/Smoke/Registration/ATF_Reregister_App_after_disconnect.lua [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+
+function Test:Close_Connection()
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", {unexpectedDisconnect = true, 
+    appID = self.applications[default_app_params.appName]})
+  self.mobileSession:Stop()
+end
+
+function Test:Reregister_Application()
+  local mobileSession = mobile_session.MobileSession(self, self.mobileConnection)
+  local on_rpc_service_started = mobileSession:StartRPC()
```
Comment (dboltovskyi) [inline test_scripts/Smoke/Registration/ATF_Reregister_App_after_disconnect.lua]: @LitvinenkoIra, Variable "on_rpc_service_started" is unused

File: test_scripts/Smoke/Registration/ATF_Reregister_App_if_two_apps_are_registered.lua [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+--     SDL->HMI: OnAppRegistered(params)
+--     SDL->appID_1: SUCCESS, success:"true":RegisterAppInterface()
+--  3. SDL assignes HMILevel after application registering:
+--     SDL->appID_1: OnHMIStatus(HMlLevel, audioStreamingState, systemContext)
+
+
+-- [[ Required Shared Libraries ]]
+local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local mobile_session = require('mobile_session')
```
Comment (dboltovskyi) [inline test_scripts/Smoke/Registration/ATF_Reregister_App_if_two_apps_are_registered.lua]: @LitvinenkoIra, "mobile_session" is unused

File: test_scripts/Smoke/Registration/ATF_Reregister_App_if_two_apps_are_registered.lua [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+      end)
+    end)
+  end)
+end
+
+commonSteps:precondition_AddNewSession()
+commonSteps:RegisterTheSecondMediaApp() 
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Reregister App if two Apps are registered")
```
Comment (dboltovskyi) [inline test_scripts/Smoke/Registration/ATF_Reregister_App_if_two_apps_are_registered.lua]: @LitvinenkoIra, Please correct description in following way (according to Atomic_template): 
```
commonFunctions:newTestCasesGroup("Test")
```

File: test_scripts/Smoke/Registration/ATF_Reregister_App_if_two_apps_are_registered.lua [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+
+function Test:Unregister_App()
+  local cid = self.mobileSession:SendRPC("UnregisterAppInterface", default_app_params1)
+  EXPECT_RESPONSE(cid, { success = true, resultCode = "SUCCESS"})
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", {unexpectedDisconnect = false, 
+                   appID = self.applications[default_app_params1]})
```
Comment (dboltovskyi) [inline test_scripts/Smoke/Registration/ATF_Reregister_App_if_two_apps_are_registered.lua]: @LitvinenkoIra, By default, when app is registered appId is stored in self.applications[app.appName]. Therefore self.applications[default_app_params1] will return nil

File: test_sets/smoke_tests.txt [hunk 18]
```diff
@@ -10,4 +10,10 @@
...
 ./test_scripts/Smoke/Resumption/ATF_Resumption_heartbeat_disconnect.lua
-./test_scripts/Smoke/Resumption/ATF_Resumption_unexpected_disconnect.lua
\ No newline at end of file
+./test_scripts/Smoke/Resumption/ATF_Resumption_unexpected_disconnect.lua
+./test_scripts/Smoke/Registration/ATF_Register_App.lua
+./test_scripts/Smoke/Registration/ATF_Register_5_session.lua
+./test_scripts/Smoke/Registration/ATF_Register_5_connection.lua
+./test_scripts/Smoke/Registration/ATF_Reregister_App.lua
+./test_scripts/Smoke/Registration/ATF_Reregister_App_after_disconnect.lua
+./test_scripts/Smoke/Registration/ATF_Reregister_App_if_two_apps_are_registered.lua
```
Comment (dboltovskyi) [inline test_sets/smoke_tests.txt]: @LitvinenkoIra, Please add new line after the last script. It is required to run the tests on CI

File: test_scripts/Smoke/Registration/ATF_Register_5_connection.lua [hunk 19]
```diff
@@ -0,0 +1,134 @@
...
+--  4th mobile device connect to system
+--  appID_4->RegisterAppInterface(params)
+--  5th mobile device connect to system
+--  appID_5->RegisterAppInterface(params)
+--
+--  Expected behavior:
+--  1. SDL successfully registers all five applications and notifies HMI and mobile
+--     SDL->HMI: OnAppRegistered(params)
+--     SDL->appID: SUCCESS, success:"true":RegisterAppInterface()
+--  3. SDL assignes HMILevel after application registering:
```
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Register_5_connection.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Register_5_connection.lua (hunk 4 above)
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Register_5_connection.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Register_5_connection.lua (hunk 2 above)
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Register_5_connection.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Register_5_connection.lua [hunk 20]
```diff
@@ -0,0 +1,134 @@
...
+--2. Device 2:
+local device2 = "192.168.100.199"
+--3. Device 3:
+local device3 = "10.42.0.1"
+--4. Device 4:
+local device4 = "1.0.0.1"
+--5. Device 5:
+local device5 = "8.8.8.8"
+
+-- Cretion dummy connections fo script
```
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Register_5_connection.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Register_5_connection.lua [hunk 21]
```diff
@@ -0,0 +1,134 @@
...
+commonFunctions:newTestCasesGroup("Preconditions")
+commonSteps:DeletePolicyTable()
+commonSteps:DeleteLogsFiles()
+
+function Test:Start_SDL()
+  self:runSDL()
+  commonFunctions:waitForSDLStart(self):Do(function()
+    self:initHMI():Do(function()
+      commonFunctions:userPrint(35, "HMI initialized")
+      self:initHMI_onReady():Do(function ()
```
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Register_5_connection.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Register_5_connection.lua [hunk 22]
```diff
@@ -0,0 +1,134 @@
...
+        commonFunctions:userPrint(35, "HMI is ready")
+      end)
+    end)
+  end)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Check that it is able to register up to 5 Apps on different connections")
```
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Register_5_connection.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Register_5_connection.lua (hunk 3 above)
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Register_5_connection.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Register_5_session.lua [hunk 23]
```diff
@@ -0,0 +1,107 @@
...
+--  Add 4 session
+--  appID_4->RegisterAppInterface(params)
+--  Add 5 session
+--  appID_5->RegisterAppInterface(params)
+--
+--  Expected behavior:
+--  1. SDL successfully registers all four applications and notifies HMI and mobile 
+--     SDL->HMI: OnAppRegistered(params)
+--     SDL->appID: SUCCESS, success:"true":RegisterAppInterface() 
+--  3. SDL assignes HMILevel after application registering:
```
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Register_5_session.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Register_5_session.lua (hunk 5 above)
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Register_5_session.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Register_5_session.lua (hunk 6 above)
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Register_5_session.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Register_App.lua (hunk 7 above)
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Register_App.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Register_App.lua (hunk 8 above)
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Register_App.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Register_App.lua (hunk 9 above)
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Register_App.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Reregister_App.lua (hunk 10 above)
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Reregister_App.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Reregister_App.lua (hunk 11 above)
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Reregister_App.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Reregister_App.lua (hunk 12 above)
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Reregister_App.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Reregister_App_after_disconnect.lua (hunk 14 above)
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Reregister_App_after_disconnect.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Reregister_App_if_two_apps_are_registered.lua (hunk 15 above)
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Reregister_App_if_two_apps_are_registered.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Reregister_App_if_two_apps_are_registered.lua (hunk 16 above)
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Reregister_App_if_two_apps_are_registered.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_sets/smoke_tests.txt (hunk 18 above)
Comment (LitvinenkoIra) [inline test_sets/smoke_tests.txt]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

File: test_scripts/Smoke/Registration/ATF_Reregister_App_if_two_apps_are_registered.lua (hunk 17 above)
Comment (LitvinenkoIra) [inline test_scripts/Smoke/Registration/ATF_Reregister_App_if_two_apps_are_registered.lua]: @dboltovskyi Done
dfd57271d04ff2169e48c2b94e2133c3b2545afb

**Pull request merged**